admin/src/react-components/service-editor.js

```diff
@@ -377,23 +410,29 @@ class ConfigurationEditor extends Component {
         </CardContent>
         <Snackbar
           anchorOrigin={{ horizontal: "center", vertical: "bottom" }}
-          open={this.state.saved || !!this.state.saveError}
+          open={this.state.saved || !!this.state.saveError || !!this.state.warningMessage}
           autoHideDuration={10000}
-          onClose={() => this.setState({ saved: false, saveError: null })}
+          onClose={() => {
+            this.setState({ saved: false, saveError: null, warningMessage: null });
+          }}
         >
           <SnackbarContent
             className={clsx({
-              [this.props.classes.success]: !this.state.saveError,
-              [this.props.classes.warning]: !!this.state.saveError
+              [this.props.classes.success]: !this.state.saveError && !this.state.warningMessage,
+              [this.props.classes.warning]: !!this.state.saveError || !!this.state.warningMessage
             })}
             message={
               <span id="import-snackbar" className={this.props.classes.message}>
                 <Icon className={clsx(this.props.classes.icon, this.props.classes.iconVariant)} />
-                {this.state.saveError || "Settings saved."}
+                {this.state.saveError || this.state.warningMessage || (this.state.saved && "Settings saved.") || ""}
               </span>
             }
             action={[
-              <IconButton key="close" color="inherit" onClick={() => this.setState({ saved: false })}>
+              <IconButton
+                key="close"
+                color="inherit"
+                onClick={() => this.setState({ saved: false, warningMessage: null })}
```

Comment:
I dont know if you want to reset the warning message, since the warning message still is valid. I could be wrong.

Reply:
I need to clear the `warningMessage` here because the user is explicitly closing the toast and the existence of the `warningMessage` is how it decides that it should be `open`. 

A related concern is that I don't actually prevent the user from saving invalid JSON here. The server will fail to parse it in that case and the themes won't be available to the client. I'm hoping that with clear instructions, people won't have too much trouble with this (and we can revisit the UI if they do).